File: OpenBanking/src/go_cardless/service/accounts_service.rs
use std::env;
use dotenvy::dotenv;
use crate::go_cardless::{ConvertsToGoCardlessAccountInfo, GoCardlessApi, GoCardlessAccountInfo};

pub struct AccountsService {
    go_cardless_secret_id: String,
    go_cardless_secret_key: String,
}

impl AccountsService {
    pub fn new() -> Self {
        dotenv().ok();

        return AccountsService {
            go_cardless_secret_id: env::var("GOCARDLESS_SECRET_ID").expect("GOCARDLESS_SECRET_ID must be set"),
            go_cardless_secret_key: env::var("GOCARDLESS_SECRET_KEY").expect("GOCARDLESS_SECRET_KEY must be set"),
        };
    }

    pub fn get_account(&self, account_id: &String) -> GoCardlessAccountInfo {
        let mut go_cardless_api = GoCardlessApi::new();

        go_cardless_api.get_token(
            self.go_cardless_secret_id.clone(),
            self.go_cardless_secret_key.clone(),
        );

        return (&go_cardless_api.get_account_info(&account_id)).to_gocardless_account_info();
    }
}

File: Services/obi_js/src/go_cardless/service/accountsService.ts
import dotenv from 'dotenv';
import { GoCardlessApi } from '../goCardlessApi';
import { GoCardlessAccountInfo } from '../model';

dotenv.config();

export class GoCardlessAccountsService {
  private readonly secretId: string;
  private readonly secretKey: string;

  constructor() {
    const secretId = process.env.GOCARDLESS_SECRET_ID;
    const secretKey = process.env.GOCARDLESS_SECRET_KEY;

    if (!secretId) {
      throw new Error('GOCARDLESS_SECRET_ID must be set');
    }
    if (!secretKey) {
      throw new Error('GOCARDLESS_SECRET_KEY must be set');
    }

    this.secretId = secretId;
    this.secretKey = secretKey;
  }

  async getAccount(accountId: string): Promise<GoCardlessAccountInfo> {
    const api = new GoCardlessApi();
    await api.getToken(this.secretId, this.secretKey);

    const accountInfo = await api.getAccountInfo(accountId);

    return {
      status: accountInfo.status,
    };
  }
}
